network/src/commonMain/kotlin/com/wire/kalium/network/KaliumKtorCustomLogging.kt

```diff
@@ -240,6 +249,49 @@ public class KaliumKtorCustomLogging private constructor(
     }
 }
 
+@Suppress("TooGenericExceptionCaught")
+private fun obfuscateAndLogMessage(text: String) {
+     try {
+        val obj = (Json.decodeFromString(text) as JsonElement)
+        if (obj.jsonArray.size > 0) {
+            obj.jsonArray.map {
+                logObfuscatedJsonElement(it)
+            }
+        } else {
+            logObfuscatedJsonElement(obj)
+        }
+        obj.toString()
+    } catch (e: Exception) {
+        "error the body content while logging "
+    }
+}
+
+fun logObfuscatedJsonElement(obj: JsonElement) {
+    obj.jsonObject.entries.toMutableSet().map {
+        when {
+            sensitiveJsonKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ******")
+            }
+            sensitiveJsonIdKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ${it.value.toString().substring(START_INDEX, END_INDEX)}")
```

Comment:
The function String.subString() can throw `StringIndexOutOfBoundsException`, it's fine if it is guarantied that all IDs are UUID and will never be shorter than 9 chars

Reply:
yeah by theory it should not be less than 9 chars, and anyway we are handling a general exception while logging these, so that should not affect the app